src/components/NotificationsDrawer/DrawerPanelContent.test.tsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import { fireEvent, render } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import configureMockStore, { MockStore } from 'redux-mock-store';
+import DrawerPanel from './DrawerPanelContent';
+import { BrowserRouter } from 'react-router-dom';
+import { markAllNotificationsAsRead, markAllNotificationsAsUnread, markNotificationAsRead, markNotificationAsUnread } from '../../redux/actions';
+import { readTestData, testData } from './notificationDrawerUtils';
+
+const mockStore = configureMockStore();
+
+const stateWithNotifications = {
+  chrome: {
+    notifications: {
+      data: testData,
+      isExpanded: true,
+      count: 3,
+    },
+  },
+};
+
+const stateWithoutNotifications = {
+  chrome: {
+    notifications: {
+      data: [],
+      isExpanded: true,
+      count: 0,
+    },
+  },
+};
+
+const stateWithReadNotifications = {
+  chrome: {
+    notifications: {
+      data: readTestData,
+      isExpanded: true,
+      count: 2,
+    },
+  },
+};
+
+const renderComponent = (store: MockStore) => {
+  return render(
+    <React.Fragment>
+      <Provider store={store}>
+        <BrowserRouter>
+          <DrawerPanel />
+        </BrowserRouter>
+      </Provider>
+    </React.Fragment>
+  );
+};
+
+describe('Drawer panel functionality', () => {
+  test('Renders the drawer panel empty successfully. ', () => {
+    const store = mockStore(stateWithoutNotifications);
+
+    const renderedResult = renderComponent(store);
+    expect(renderedResult.getByText('Notifications')).toBeInTheDocument();
+  });
+
+  test('Renders notification drawer with notifications successfully', () => {
+    const store = mockStore(stateWithNotifications);
+
+    const renderedResult = renderComponent(store);
+    expect(renderedResult.getByText('Test Notification 1')).toBeInTheDocument();
+  });
+
+  test('Marking notification as read successfully', () => {
+    const store = mockStore(stateWithNotifications);
+
+    const renderedResult = renderComponent(store);
+
+    const checkbox = renderedResult.getAllByRole('checkbox');
+    fireEvent.click(checkbox[0]);
```

Comment:
Events typically have to be wrapped in `act` or `waitFor` or their async variants. Otherwise it will output a lot of errors in console and state transitions might not happen.

```
  console.error
    Warning: An update to Popper inside a test was not wrapped in act(...).
    
    When testing, code that causes React state updates should be wrapped into act(...):
    
    act(() => {
      /* fire events that update state */
    });
    /* assert on the output */
    
    This ensures that you're testing the behavior the user would see in the browser. Learn more at https://reactjs.org/link/wrap-tests-with-act
        at Popper (/home/martin/insights/insights-chrome/node_modules/@patternfly/react-core/src/helpers/Popper/Popper.tsx:152:3)
        at DropdownBase (/home/martin/insights/insights-chrome/node_modules/@patternfly/react-core/src/components/Dropdown/Dropdown.tsx:67:3)
        at Dropdown
        at div
        at div
        at NotificationDrawerHeader (/home/martin/insights/insights-chrome/node_modules/@patternfly/react-core/src/components/NotificationDrawer/NotificationDrawerHeader.tsx:30:3)
        at div
        at NotificationDrawerBase (/home/martin/insights/insights-chrome/node_modules/@patternfly/react-core/src/components/NotificationDrawer/NotificationDrawer.tsx:16:3)
        at NotificationDrawer
        at innerRef (/home/martin/insights/insights-chrome/src/components/NotificationsDrawer/DrawerPanelContent.tsx:58:28)
        at basenameProp (/home/martin/insights/insights-chrome/node_modules/react-router/lib/components.tsx:378:13)
        at basename (/home/martin/insights/insights-chrome/node_modules/react-router-dom/index.tsx:354:3)
        at Provider (/home/martin/insights/insights-chrome/node_modules/react-redux/lib/components/Provider.js:19:3)

```

Reply:
Oh interesting, that's true for every "fireEvent" and things of that nature  i use?

I'll go over them

Reply:
It's not about the `fireEvent` specifically. It's about whatever triggers a state transition in the component. Usually, these are events, but they also can be promises or even when component mounts. But in this case, it is likely its caused by the fireEvent calls.